src/core/SpyFactory.js

```diff
@@ -1,45 +1,61 @@
 getJasmineRequireObj().SpyFactory = function(j$) {
 
-  function SpyFactory(getCustomStrategies) {
-    var self = this;
-
-    this.createSpy = function(name, originalFn) {
-      return j$.Spy(name, originalFn, getCustomStrategies());
-    };
-
-    this.createSpyObj = function(baseName, methodNames) {
-      var baseNameIsCollection = j$.isObject_(baseName) || j$.isArray_(baseName);
-
-      if (baseNameIsCollection && j$.util.isUndefined(methodNames)) {
-        methodNames = baseName;
-        baseName = 'unknown';
-      }
-
-      var obj = {};
-      var spiesWereSet = false;
-
-      if (j$.isArray_(methodNames)) {
-        for (var i = 0; i < methodNames.length; i++) {
-          obj[methodNames[i]] = self.createSpy(baseName + '.' + methodNames[i]);
-          spiesWereSet = true;
-        }
-      } else if (j$.isObject_(methodNames)) {
-        for (var key in methodNames) {
-          if (methodNames.hasOwnProperty(key)) {
-            obj[key] = self.createSpy(baseName + '.' + key);
-            obj[key].and.returnValue(methodNames[key]);
-            spiesWereSet = true;
-          }
-        }
-      }
-
-      if (!spiesWereSet) {
-        throw 'createSpyObj requires a non-empty array or object of method names to create spies for';
-      }
-
-      return obj;
-    };
-  }
-
-  return SpyFactory;
+    function SpyFactory(getCustomStrategies) {
+        var self = this;
+
+        this.createSpy = function(name, originalFn) {
+            return j$.Spy(name, originalFn, getCustomStrategies());
+        };
+
+        this.createSpyObj = function(baseName, methodNames, propertyNames) {
+            var baseNameOrMethodNamesAreCollections = j$.isObject_(baseName) || j$.isArray_(baseName) || j$.isObject_(propertyNames) || j$.isArray_(propertyNames);
```

Comment:
It looks like this variable now also has some knowledge about the state of the propertyNames as well. I'd like to see it renamed or lose that knowledge.

Reply:
This was a mistake, it should have been methodNames insteadof propertyNames

Reply:
This wasn't even needed. A change in the line 13 does the trick.